### PR TITLE
Validate fhir with ije

### PR DESCRIPTION
**Pull request by smacadam**
PR for [NVSS-91](https://tracker.codev.mitre.org/browse/NVSS-91)  
Added a new canary feature to validate a FHIR record using an arbitrary IJE file  

To test, run canary and click the "Validate FHIR Death Records (Producing)". Upload an ije file in the first input and a fhir file in the second. Check that the results of the test are as expected and you can download the results file. Visually check that the UI is consistent with the rest of canary.

### CODE REVIEW

File: canary/ClientApp/src/components/dashboard/Dashboard.js [hunk 1]
```diff
@@ -72,6 +72,12 @@ export class Dashboard extends Component {
                   description="Test a data provider system's ability to produce pre-defined records as tested at Connectathons."
                   route="test-connectathon-dash/records"
                 />
+                 <DashboardItem
+                  icon="tasks"
+                  title="Validate FHIR Death Records (Producing)"
```
Comment (rbclark) [inline canary/ClientApp/src/components/dashboard/Dashboard.js:77]: Should this also be called `Syntax Checker` and follow the rules we laid out today (FHIR Death Records -> FHIR VRDR Records)?

Also the fact that this checks against an IJE seems important enough to call out in the title right?

File: canary/ClientApp/src/components/tests/FHIRIJEValidatorProducing.js [hunk 2]
```diff
@@ -0,0 +1,216 @@
...
+                    <Icon name="upload" />
+                    <Header.Content>
+                      Step 1: Upload IJE Record
+                      <Header.Subheader>
+                        Upload the IJE record you will use to validate the FHIR record. The below prompt allows you to paste the record, upload it as a file, or POST to Canary.
+                      </Header.Subheader>
+                    </Header.Content>
+                  </Header>
+                  <div className="p-b-15">
+                    <Getter name="upload-ije" id="upload1" updateRecord={this.updateRecord} ijeOnly noFormat />
```
Comment (rbclark) [inline canary/ClientApp/src/components/tests/FHIRIJEValidatorProducing.js:160]: Normally when the user submits a file we show a success or fail message below so it is clear that they are able to continue. That may be useful on both the upload boxes.


Screen Shot 2021-06-07 at 5 47 52 PM

**Review by rbclark — CHANGES_REQUESTED** [review]
I generated a synthetic death record and converted it using our converter over to IJE. When I compared the two I still got a failure (I've attached the two files)


[record-1623102692637.MOR.txt](https://github.com/nightingaleproject/canary/files/6612217/record-1623102692637.MOR.txt)
[record-1623102655596.xml.txt](https://github.com/nightingaleproject/canary/files/6612218/record-1623102655596.xml.txt)

**Comment by smacadam** [comment]
For the synthetic data issue - it looks like the DeathRecordFaker is using a different NullFlavor reference than everywhere else in the code. 
DeathRecordFaker.cs uses "http://hl7.org/fhir/v3/NullFlavor"
Everywhere else uses "http://terminology.hl7.org/CodeSystem/v3-NullFlavor"

The http://hl7.org/fhir/v3/NullFlavor url doesn't resolve, so seems like the fix should be to update the url in DeathRecordFaker to http://terminology.hl7.org/CodeSystem/v3-NullFlavor.

**Comment by pkmitre** [comment]
Updating DeathRecordFaker seems like the right approach

File: canary/ClientApp/src/components/dashboard/Dashboard.js (hunk 1 above)
Comment (smacadam) [reply canary/ClientApp/src/components/dashboard/Dashboard.js:77]: In this case, it is checking whether the values are correct so I think it is more of a validator. Doesn't the syntax check just check for correct formatting? I will add in IJE to the title and change it to VRDR Records

**Pull request merged**